client/analytics/page-view-tracker/component.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { PropTypes } from 'react';
+
+export const PageViewTracker = recorder => React.createClass( {
+	getInitialState: () => ( {
+		timer: null
+	} ),
+
+	componentDidMount() {
+		this.queuePageView();
+	},
+
+	componentWillUnmount() {
+		clearTimeout( this.state.timer );
+	},
+
+	queuePageView() {
+		const {
+			delay = 0,
+			path,
+			title
+		} = this.props;
+
+		if ( this.state.timer ) {
+			return;
+		}
+
+		if ( ! delay ) {
+			return this.setState(
+				{ timer: true },
+				recorder( path, title )
+			);
```

Comment:
that's a good point. since this happens in `componentDidMount()` it should only ever be called once. setting the state was intended as a safety but I don't think it adds any real safety